### PR TITLE
FEATURE: add daysOff hours to week hours and timeBadge

### PR DESCRIPTION
**Pull request by o-mehlii**
https://trello.com/c/hu4HNZ5R/361-add-sickness-and-vacation-in-total-hours-in-calendar
https://ukad.slack.com/archives/C069N5LUP3M/p1708697779970829
plus fixed weekday definition, getWeekNumber function was not working correctly for leap years

### CODE REVIEW

File: renderer/src/shared/TimeBadge/TimeBadge.tsx [hunk 1]
```diff
@@ -20,8 +38,8 @@ export default function TimeBadge({ hours, startTime, selectedDate }: TimeBadgeP
     );
   }
   if (
-    (hours < 8 && selectedDate.toLocaleDateString() !== curDate.toLocaleDateString()) ||
-    (hours < 8 && curTime - stringToMinutes(startTime) > 480)
+    (hours + dayOffDuration < 8 && selectedDate.toLocaleDateString() !== curDate.toLocaleDateString()) ||
+    (hours + dayOffDuration < 8 && curTime - stringToMinutes(startTime) > 480)
```
Comment (okhmelevskyi) [inline renderer/src/shared/TimeBadge/TimeBadge.tsx:42]: Could you, please, move 8 and 480 numbers to constants and call with a sense? 
Ex:
```
export const NAME_CONST = 8;
export const NAME_CONST  = 480;
```

**Review by okhmelevskyi — APPROVED** [review]

File: renderer/src/components/Calendar/FullCalendarWrapper.tsx [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
-import { isTheSameDates } from "@/helpers/utils/datetime-ui";
+import { isTheSameDates, MS_PER_HOUR, getWeekNumber } from "@/helpers/utils/datetime-ui";
 import { formatDuration } from "@/helpers/utils/reports";
 import { DatePointApi, DayCellContentArg, EventContentArg, WeekNumberContentArg } from "@fullcalendar/core";
 import { CalendarDaysIcon, FaceFrownIcon, GlobeAltIcon } from "@heroicons/react/24/outline";
 import { ExclamationCircleIcon } from "@heroicons/react/24/solid";
 import { DayOff, FormattedReport, FullCalendarWrapperProps } from "./types";
 
+type daysOffAccumulatorType = { numberedDays: string[]; hours: number };
+
```
Comment (a-nahaitsev) [inline renderer/src/components/Calendar/FullCalendarWrapper.tsx:10]: move type to types file

File: renderer/src/helpers/utils/datetime-ui.ts [hunk 3]
```diff
@@ -36,8 +36,10 @@ export function getWeekNumber(dateString: string) {
   const dateObj = getDateFromString(dateString);
   const startOfYear = new Date(dateObj.getFullYear(), 0, 1);
   const days = Math.floor((dateObj.getTime() - startOfYear.getTime()) / (24 * 60 * 60 * 1000));
+  const year = dateObj.getFullYear();
+  const leapYearDay = year === 2020 || year === 2024 || year === 2028 || year === 2032 ? 0 : 1;
```
Comment (a-nahaitsev) [inline renderer/src/helpers/utils/datetime-ui.ts:40]: You harcoded years, right? could you create a function to calculate a leap year?

File: renderer/src/shared/TimeBadge/TimeBadge.tsx (hunk 1 above)
Comment (a-nahaitsev) [reply renderer/src/shared/TimeBadge/TimeBadge.tsx:43]: and 6 from 29-30 lines as well

**Pull request merged**